internal/services/cdn/cdn_frontdoor_helpers.go

```diff
@@ -439,6 +484,49 @@ func updateRouteAssociations(d *pluginsdk.ResourceData, meta interface{}, routeI
 	return nil
 }
 
+func setRouteRuleSetAssociations(d *pluginsdk.ResourceData, meta interface{}, id *parse.FrontDoorRuleSetAssociationId, ruleSets []interface{}, props *cdn.RouteProperties, errorTxt string, futureErrorTxt string) error {
+	client := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
+	workaroundsClient := azuresdkhacks.NewCdnFrontDoorRoutesWorkaroundClient(client)
+	ctx, routeCancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer routeCancel()
+
+	// NOTE: You need to always pass these these three on update else you will
+	// disable/disassociate your custom domains, cache configuration or rule sets...
+	updateProps := azuresdkhacks.RouteUpdatePropertiesParameters{
+		CustomDomains:      props.CustomDomains,
+		CacheConfiguration: props.CacheConfiguration,
+	}
+
+	// Since delete is the only function that passes nil as the ruleSet value
+	// we know it is a delete operation...
+	if ruleSets != nil || len(ruleSets) != 0 {
+		// associate only the rule sets contained within the association resource
+		updateProps.RuleSets = expandRuleSetReferenceArray(ruleSets)
+	} else {
+		// Disassociate all rule sets from the route
+		updateProps.RuleSets = nil
+	}
+
+	updateParams := azuresdkhacks.RouteUpdateParameters{
+		RouteUpdatePropertiesParameters: &updateProps,
+	}
+
+	// lock the route resource for update...
+	locks.ByName(id.AssociationName, cdnFrontDoorRouteResourceName)
+	defer locks.UnlockByName(id.AssociationName, cdnFrontDoorRouteResourceName)
+
+	future, err := workaroundsClient.Update(ctx, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName, updateParams)
```

Comment:
as discussed in the sync, `PATCH` API's (`Update` calls in the SDK) will be delta/patch API's, which is to say that if the JSON field is omitted then it won't be touched - however the `PUT`/`POST` API's (`CreateOrUpdate` calls in the SDK) should remove these fields without the need for the `azuresdkhacks` folder) - so whilst this works fine for now, perhaps that'd make life easier in the future?

Reply:
I was planning to do the move to the new SDK(e.g. `go-azure-sdk`) once the new Front Door API GA's, for now it is still stuck on the track1 `azure-sdk-for-go` so the `azuresdkhacks` are currently needed for now. 🙁

Reply:
Using the POST/PUT rather than the PATCH will work in either track1 or `go-azure-sdk` - so we can update this to use the `CreateOrUpdate` call today and remove this workaround

Reply:
Fixed.